contracts/starknet/ds_value.cairo

```diff
@@ -0,0 +1,47 @@
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin, BitwiseBuiltin
+from starkware.cairo.common.uint256 import Uint256
+
+// bool    has;
+@storage_var
+func _has() -> (res: felt) {
+}
+// bytes32 val;
+@storage_var
+func _val() -> (res: Uint256) {
+}
+
+// function peek() public view returns (bytes32, bool) {
+//     return (val,has);
+// }
+@view
+func peek{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (
+    val: Uint256, has: felt
+) {
+    let (val) = _val.read();
+    let (has) = _has.read();
+    return (val, has);
+}
+
+// function read() public view returns (bytes32) {
```

Comment:
No read implementation?

Reply:
Was not necessary to `mock` but I can add it for full feature parity for sure.